### ISSUE TITLE
does it support character level embedding?

### ISSUE THREAD

**Issue by zxu7** [tensorflow/nmt#152]
If yes, how do I turn it on during training?

**Comment by oahziur** [comment]
@zxu7 The code doesn't have a character level embedding option.

However, you may tokenize data at character level, and prepare a character level vocab file to train a character model with the codebase.

**Comment by AlkaSaliss** [comment]
Hi @oahziur ,
I'm trying to train a character level NMT mode l. So I bascially built a vocabulary consisting of characters
![char_vocab](https://user-images.githubusercontent.com/33458274/43587584-cfc9f01c-966a-11e8-8177-3e11b88a3ad8.png)
But the problem I'm facing is that the `space`  character is also part of my vocabulary but if I add it to the vocab file I get an error:
![vocab_error](https://user-images.githubusercontent.com/33458274/43587807-591f0d66-966b-11e8-8d06-6c033e014f63.png)
So I tried to replace the `space` with a special character that is not in my vocab, say `~`
But during the training the model keep predicting the unknown token `<unk>`, implying that there may be a tokenization problem. My question is how should I tokenize the data so that the model work with the character-level vocabulary ?
As for now I let my data in the same format as for word-level models
- source file: 
![src_exple](https://user-images.githubusercontent.com/33458274/43588620-2eff7bea-966d-11e8-9c02-d1fbcc8bd34d.png)

- target file: 
![tgt_exple](https://user-images.githubusercontent.com/33458274/43588637-3915d0d4-966d-11e8-9fbd-94b558ebdafa.png)

thanks!

**Comment by eswarjal09** [comment]
Hi @oahziur ,
To train it as a character level model, i should change the delimiter space to empty string in the tensorflow string_split file or is there a way i can do that in the nmt code?

**Comment by shanalikhan** [comment]
@AlkaSaliss were you able to generate character embedding and run NMT using for character level?

**Comment by AlkaSaliss** [comment]
@shanalikhan Yes I managed to get it work with character-level vocabulary. See my comment above. I'm not sure it is the best way but could work as workaround.
